### ISSUE TITLE
Memory usage increasing predictions with dropout enabled

### ISSUE THREAD

**Issue by eyesonlyhack** [keras-team/keras#10338]
I have been running code for doing predictions with uncertainty for a number of models and have noticed an issue of my memory usage just continually going up for no obvious reason.

Basically I am performing a large number of predictions with the dropout layers enabled. I am using tensorflow as a backend. This method is described in #9412 

I have created a minimal version of this problem:
https://gist.github.com/eyesonlyhack/e676a0e2230a5a09d31aa3401a59d009

You can probably call K.clear_session to free up this memory but then you would have to rebuild/reload the model after every clear which seems wasteful.

For interest sake, I switched to mxnet for my Keras backend and the problem is goes away.

**Comment by eyesonlyhack** [comment]
Found the issue. I was because a creates the K.function for each call, this creates ops so the memory goes up, ops are not collected as designed. This behavior is inconsistent with mxnet which does not have this issue. Guess this might be out of scope to fix or is not possible to make the behavior consistent?

**Comment by Dref360** [comment]
Hey, as I said on Slack, this is as designed for the Tensorflow backend, you can't delete ops, so we won't fix.

**Comment by chjq201410695** [comment]
Does this issue be solved?

**Comment by wangyexiang** [comment]
> I have been running code for doing predictions with uncertainty for a number of models and have noticed an issue of my memory usage just continually going up for no obvious reason.
> Basically I am performing a large number of predictions with the dropout layers enabled. I am using tensorflow as a backend. This method is described in #9412
> I have created a minimal version of this problem:
> https://gist.github.com/eyesonlyhack/e676a0e2230a5a09d31aa3401a59d009
> You can probably call K.clear_session to free up this memory but then you would have to rebuild/reload the model after every clear which seems wasteful.
> For interest sake, I switched to mxnet for my Keras backend and the problem is goes away.
how to switch to mxnet for  Keras backend？？

**Comment by wangyexiang** [comment]
how to switch to mxnet for Keras backend？？ @eyesonlyhack

**Comment by eyesonlyhack** [comment]
@wangyexiang Try follow this tutorial: https://medium.com/@julsimon/apache-mxnet-as-a-backend-for-keras-2-9993f97843e7

**Comment by wangyexiang** [comment]
Thank you. @eyesonlyhack
